airflow/providers/mongo/hooks/mongo.py

```diff
@@ -90,15 +84,19 @@ def get_conn(self) -> MongoClient:
             options.update({"ssl_cert_reqs": CERT_NONE})
 
         self.client = MongoClient(self.uri, **options)
-
         return self.client
 
-    def close_conn(self) -> None:
-        """Closes connection"""
-        client = self.client
-        if client is not None:
-            client.close()
-            self.client = None
+    def create_uri(self) -> str:
+        """
+        Create URI string from the given credentials.
+
+        :return: URI string.
+        """
+        srv = self.extras.pop("srv", False)
+        scheme = "mongodb+srv" if srv else "mongodb"
+        creds = f"{self.connection.login}:{self.connection.password}@" if self.connection.login else ""
+        port = f":{self.connection.port}" if self.connection.port else ""
+        return f"{scheme}://{creds}{self.connection.host}{port}/{self.connection.schema}"
```

Comment:
This might be not a main idea of this PR but also would be nice if we url-encode login and password:
https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html#pymongo.mongo_client.MongoClient

Reply:
Done!